sdks/go/pkg/beam/core/metrics/store.go

```diff
@@ -128,9 +128,11 @@ func (e Extractor) ExtractFrom(store *Store) error {
 		}
 	}
 	if e.MsecsInt64 != nil {
+		store.mu.Lock()
```

Comment:
This isn't required, you're already acquiring locks up at the start of the method. (the RLock & RUnlock since this only reads data, not writes to the store) That's causing the deadlock failures.

Reply:
Got it.